packages/cdktf-cli/bin/cmds/terraform-check.ts

```diff
@@ -1,9 +1,27 @@
 import { Terraform } from './ui/models/terraform'
+import * as semver from 'semver';
+
+const MIN_SUPPORTED_VERSION = '0.12.0'
+const VERSION_REGEXP = /Terraform v\d+.\d+.\d+/
 
 export const terraformCheck = async (): Promise<void> => {
   try {
-    const terraform = new Terraform('./')
-    await terraform.version()
+    const terraform: Terraform = new Terraform('./')
+
+    const terraformVersion = await terraform.version()
+    const terraformVersionMatches = terraformVersion.match(VERSION_REGEXP)
+
+    if ( terraformVersionMatches !== null) {
+
+      // Should always be the first match found in the string
+      let cleanTerraformVersion = semver.clean(terraformVersionMatches[0].substring(terraformVersionMatches[0].indexOf('v')))
```

Comment:
@skorfmann I've explicitly parsed the string so it becomes something like `Terraform v13.0.1` which can then be compared against by semver.